### PR TITLE
update(demo): Enhance the list of Vectara demos and applications

### PR DESCRIPTION
**Pull request by pwoznic**
Updated the topic with a table of our latest demos and applications, e.g. Vectara Agentic

### CODE REVIEW

**Comment by ofermend** [comment]
Paul,
From the diff I see both actual demos (listed under "https://www.vectara.com/developers/build/demos") and sample code (listed under "https://www.vectara.com/developers/build/sample-code") in the same table. Can you please clarify why?
I think of these as typiclaly two different types of resources (and thus the two separate pages in the website) - one is actual demos and the other is code you can look at (in repos, like vectara-ingest, vectara-answer, ragtime, etc).

Also, is the intent to list ALL demos and sample code or just a sample?
I'm concerned that if we try ALL of them it'd be hard to follow, so maybe just list the most important ones?

**Comment by cjcenizal** [comment]
We have pages dedicated to listing this stuff on vectara.com. If we avoid duplicating information we will make maintenance easier and also make the experience of locating this information less confusing for folks. Can we cross-link to the appropriate pages on vectara.com instead of duplicating that info?

**Comment by pwoznic** [comment]
> Paul, From the diff I see both actual demos (listed under "https://www.vectara.com/developers/build/demos") and sample code (listed under "https://www.vectara.com/developers/build/sample-code") in the same table. Can you please clarify why? I think of these as typiclaly two different types of resources (and thus the two separate pages in the website) - one is actual demos and the other is code you can look at (in repos, like vectara-ingest, vectara-answer, ragtime, etc).
> 
> Also, is the intent to list ALL demos and sample code or just a sample? I'm concerned that if we try ALL of them it'd be hard to follow, so maybe just list the most important ones?

The original topic had a few of the "Ask" apps and Vectara ingest/answer. I added Agentic and then the new ones from https://www.vectara.com/developers/build/demos. I can move out the sample code ones and stick to applications (or create a separate page). My intent was to show everything we have as of today because the original topic was outdated and we have added many more since. I can reduce the size of the list and then just link to the site if that works.

**Comment by pwoznic** [comment]
> We have pages dedicated to listing this stuff on vectara.com. If we avoid duplicating information we will make maintenance easier and also make the experience of locating this information less confusing for folks. Can we cross-link to the appropriate pages on vectara.com instead of duplicating that info?

Sure, I can just link to https://www.vectara.com/developers/build/demos instead of links to each app since our site already has those links. The intent was to update this outdated topic to showcase the latest list.

File: www/docs/sample-vectara-apps.md [hunk 1]
```diff
@@ -6,45 +6,51 @@ sidebar_label: Sample Demos and Applications
...
 import {Config} from './definitions.md';
 
-Explore the <Config v="names.product"/> platform through various 
-demo applications that demonstrates the power of GenAI conversational 
-search and Retrieval Augmented Generation (RAG). 
+Discover the <Config v="names.product"/> platform's potential with our diverse collection of 
+open-source demo applications. These examples showcase the power of generative 
+AI applications and Retrieval Augmented Generation (RAG) in action, 
+inspiring developers to innovate and integrate cutting-edge capabilities into 
+their own solutions.
```
Comment (omnomagonz) [inline www/docs/sample-vectara-apps.md:13]:
```suggestion
<Config v="names.product"/>'s platform enables you to build generative AI and Retrieval 
Augmented Generation (RAG) applications quickly and without expertise in AI or 
ML concepts. We have a diverse set of open-source demo applications that 
showcase the power of generative AI applications for realistic use cases. 
These demo applications are just a sample of what developers can create with 
<Config v="names.product"/>.

```

Non-blocking suggestion. My personal preference when reading docs is to see language that's more direct and technical vs. attempting to convince me of something. I'm not sure my own suggestion does enough to meet this criteria, to be honest.

Words like "Discover", "inspiring", etc seem to lean more into the marketing and conversion aspect vs how the product works and why a reader may be interested in it, but this is entirely anecdotal and personal preference. Feel free to discard this suggestion if you disagree @pwoznic 😄

File: www/docs/sample-vectara-apps.md [hunk 2]
```diff
@@ -6,45 +6,51 @@ sidebar_label: Sample Demos and Applications
...
+various industry challenges. From question-answering tools that provide 
+precise documentation support, to advanced AI agents which demonstrate agentic 
+capabilities for handling complex tasks, these applications cater to diverse 
+developer needs.
+
+Explore chatbots that leverage conversational search for real-time 
+interactions, virtual agents that automate workflows, and custom data 
+ingestion tools that simplify integrating RAG capabilities. Each demo serves 
+as a practical example to inspire developers to build robust, AI-enhanced 
+solutions tailored to their unique use cases.
```
Comment (omnomagonz) [inline www/docs/sample-vectara-apps.md:25]:
```suggestion
Our demo applications cover a variety of use cases that address known industry  
challenges. They include question-answering tools for documentation support,  
AI agents for complex tasks, chatbots that use conversational search for  
real-time interactions, virtual agents that automate workflows, and custom  
data ingestion tools for integrating RAG capabilities into an existing  
application. Each demo serves as a practical example to inspire developers  
to build AI-enhanced solutions for their specific use cases.  
```

Non-blocking: these two paragraphs seemed to be saying the same/similar things as each other, but also the first paragraph. I thought combining them and removing some of the overlap might be helpful

**Review by omnomagonz — COMMENTED** [review]
Two non-blocking suggestions @pwoznic 😄

**Review by omnomagonz — APPROVED** [review]

**Review by ofermend — APPROVED** [review]
LGTM
